test/images/agnhost/README.md

```diff
@@ -72,8 +62,213 @@ created with the `pause` argument instead, allowing us execute multiple commands
+valid IP addresses:
+
+- `POD_IP`
+- `NODE_IP`
+
+Serves the following endpoints on the given port (defaults to `8080`).
+
+- `/whoami` - returns the request's IP address.
+- `/checknosnat` - queries  `ip/whoami` for each provided IP (`/checknosnat?ips=ip1,ip2`),
+  and if all the response bodies match the `POD_IP`, it will return a 200 response, 500 otherwise.
+
+Usage:
+
+```console
+    kubectl run test-agnhost \
+      --generator=run-pod/v1 \
+      --image=gcr.io/kubernetes-e2e-test-images/agnhost:1.1 \
+      --restart=Never \
+      --env "POD_IP=<POD_IP>" \
+      --env "NODE_IP=<NODE_IP>" \
+      -- no-snat-test [--port <port>]
+```
+
+
+### no-snat-test-proxy
+
+Serves the `/checknosnat` endpoint on the given port (defaults to `31235`). The endpoint
+proxies the request to the given `target` (`/checknosnat?target=target_ip&ips=ip1,ip2`
+-> `target_ip/checknosnat?ips=ip1,ip2`) and will return the same status as the status
+as the proxied request, or 500 on error.
+
+
+Usage:
+
+```console
+    kubectl exec test-agnhost -- /agnhost no-snat-test-proxy [--port <port>]
+```
+
+
+### pause
+
+It will pause the execution of the binary. This can be used for containers
+which have to be kept in a `Running` state for various purposes, including
+executing other `agnhost` commands.
+
+Usage:
+
+```console
+    kubectl exec test-agnhost -- /agnhost pause
+```
+
+
+### port-forward-tester
+
+Listens for TCP connections on a given address and port, optionally checks the data received,
+and sends a configurable number of data chunks, with a configurable interval between chunks.
+
+The subcommand is using the following environment variables:
+
+- `BIND_ADDRESS` (optional): The address on which it will start listening for TCP connections (default value: `localhost`)
+- `BIND_PORT`: The port on which it will start listening for TCP connections.
+- `EXPECTED_CLIENT_DATA` (optional): If set, it will check that the request sends the same exact data.
+- `CHUNKS`: How many chunks of data to write in the response.
+- `CHUNK_SIZE`: The expected size of each written chunk of data. If it does not match the actual size of the written data, it will exit with the exit code `4`.
+- `CHUNK_INTERVAL`: The amount of time to wait in between chunks.
+
+Usage:
+
+```console
+    kubectl run test-agnhost \
+      --generator=run-pod/v1 \
+      --image=gcr.io/kubernetes-e2e-test-images/agnhost:1.1 \
+      --restart=Never \
+      --env "BIND_ADDRESS=localhost" \
+      --env "BIND_PORT=8080" \
+      --env "EXPECTED_CLIENT_DATA='Hello there!'" \
+      --env "CHUNKS=1" \
+      --env "CHUNK_SIZE=10" \
+      --env "CHUNK_INTERVAL=1" \
+      -- port-forward-tester
+```
+
+
 ## Image
 
-The image can be found at `gcr.io/kubernetes-e2e-test-images/agnhost:1.0` for Linux
-containers, and `e2eteam/agnhost:1.0` for Windows containers. In the future, the same
+The image can be found at `gcr.io/kubernetes-e2e-test-images/agnhost:1.1` for Linux
```

Comment:
```suggestion
The image can be found at `gcr.io/kubernetes-e2e-test-images/agnhost:2.0` for Linux
```

Reply:
Oups, I forgot to update the version in the documentation.